bytecomp/matching.ml

```diff
@@ -2440,6 +2440,9 @@ let combine_constructor loc arg ex_pat cstr partial ctx def
           with
           | (1, 1, [0, act1], [0, act2]) ->
               Lifthenelse(arg, act2, act1)
+          | (2,0, [(i1,act1); (_,act2)],[]) ->
+            if i1 = 0 then Lifthenelse(arg, act2, act1)
```

Comment:
This is a change in the generated code from `Lifthenelse(arg != 0, ...)` to `Lifthenelse(arg, ...)` which applies to booleans as well as any variant with 2 const cases.
This looks OK as it seems consistent with how the back-end inteprets `Lifthenelse`, e.g. in the case just above this one.